components/asic/asic.c

```diff
@@ -0,0 +1,76 @@
+#include "asic.h"
```

Comment:
I think we should rename this to `asic/task.c` - :+1: ? :-1: ?

Reply:
on think is that incldues will be 'task.h' which will be confusing - so maybe asic_task?

Reply:
kk - for a follow up PR :heavy_check_mark: